projects/observability/src/shared/dashboard/widgets/top-n/top-n-widget-renderer.component.ts

```diff
@@ -63,11 +63,15 @@ export class TopNWidgetRendererComponent extends InteractiveDataWidgetRenderer<T
     private readonly entityNavService: EntityNavigationService
   ) {
     super(api, changeDetector);
-    this.setInitialMetricSpecification();
-    this.setOptionsObservables();
+
+    this.model.getData().subscribe(fetcher => {
```

Comment:
This should move into the tap that was removed in the `fetchData` function. Otherwise, the getData call there isn't ever being used (and we'd be calling getData indiscriminately, regardless of whether it's ultimately used).